Toggl.Core.UI/ViewModels/MainTabBarViewModel.cs

```
@@ -1,153 +1,107 @@
+                dependencyContainer.AnalyticsService,
+                dependencyContainer.SchedulerProvider,
+                dependencyContainer.RxActionFactory));
+
+            calendarViewModel = new Lazy<ViewModel>(() => new CalendarViewModel(
+                dependencyContainer.DataSource,
+                dependencyContainer.TimeService,
+                dependencyContainer.UserPreferences,
+                dependencyContainer.AnalyticsService,
+                dependencyContainer.BackgroundService,
+                dependencyContainer.InteractorFactory,
+                dependencyContainer.OnboardingStorage,
+                dependencyContainer.SchedulerProvider,
+                dependencyContainer.PermissionsChecker,
+                dependencyContainer.NavigationService,
+                dependencyContainer.RxActionFactory));
+
+            settingsViewModel = new Lazy<ViewModel>(() => new SettingsViewModel(
+                dependencyContainer.DataSource,
+                dependencyContainer.SyncManager,
+                dependencyContainer.PlatformInfo,
+                dependencyContainer.UserPreferences,
+                dependencyContainer.AnalyticsService,
+                dependencyContainer.InteractorFactory,
+                dependencyContainer.OnboardingStorage,
+                dependencyContainer.NavigationService,
+                dependencyContainer.RxActionFactory,
+                dependencyContainer.PermissionsChecker,
+                dependencyContainer.SchedulerProvider));
+
+            Tabs = getViewModels().ToImmutableList();
         }
 
-        public override async Task Initialize()
+        public TViewModel GetViewModel<TViewModel>()
+            where TViewModel : class, IViewModel
         {
-            await base.Initialize();
+            var expectedType = typeof(TViewModel);
+            if (expectedType == typeof(MainViewModel))
+                return mainViewModel.Value as TViewModel;
```

Turn these into explicit casts so it fails fast on null.